boofuzz/monitors/network_monitor.py

```diff
@@ -0,0 +1,55 @@
+from .imonitor import IMonitor
+from . import pedrpc
+
+# Important: IMonitor needs to come *before* pedrpc.Client in the
+# Inheritance list for the method resolution order to produce
+# correct results.
+
+
+class NetworkMonitor(IMonitor, pedrpc.Client):
+    """
+    Proxy class for the network monitor interface.
+
+    In Versions < 0.2.0, boofuzz had network and process monitors
+    that communicated over RPC. The RPC client was directly passed
+    to the session class, and resolved all method calls dynamically
+    on the RPC partner.
+
+    Since 0.2.0, every monitor class must implement the abstract class
+    IMonitor, which defines a common interface among all Monitors. To
+    aid future typehinting efforts and to make Network- and Process Monitors
+    disambiguable, this explicit proxy class has been introduced that
+    fast-forwards all calls to the RPC partner.
+
+    .. versionadded:: 0.2.0
+    """
+
+    def __init__(self, host, port):
+        super(IMonitor, self).__init__()
+        super(pedrpc.Client, self).__init__(host, port)
```

Comment:
When I try to instantiate `NetworkMonitor` I get some errors.
Not completely sure but I don't think this is how super works.

Either way, the static code analysis of PyCharm was once more right about something being wrong here :)

Also, `alive()` is missing.

Reply:
*sigh* yeah, probably. Inheritance in python is a mess at times. I think I got it down now, at least the method resoltution order that python 3 outputs seems to be what I expect and making the bases even more explicit seems to work at least in my iPython shell. Let's see.